db/mongo/client.go

```diff
@@ -47,47 +46,46 @@ func (c client) SaveEmployee(employee *models.Employee) (string, error) {
 	opts := options.Update().SetUpsert(true)
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
 	if _, err := collection.UpdateOne(context.TODO(), bson.M{"id": employee.ID}, bson.M{"$set": employee}, opts); err != nil {
-		panic("oops cannot update employee")
+		return "", errors.Wrap(err, "failed to add/update employee")
 	}
-	return "", nil
+	return employee.ID, nil
 }
 
 func (c client) GetEmployeeByID(id string) (*models.Employee, error) {
 	var employee *models.Employee
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
 	if err := collection.FindOne(context.TODO(), bson.M{"id": id}).Decode(&employee); err != nil {
 		if errors.Is(err, mongo.ErrNoDocuments) {
-			panic("oops error occurred ")
+			return nil, errors.Wrap(err, "failed to get employee")
 		}
-		return nil, err
 	}
 	return employee, nil
 }
 
-func (c client) DeleteEmployee(id string) error {
+func (c client) DeleteEmployee(id string) (string, error) {
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
 	if _, err := collection.DeleteOne(context.TODO(), bson.M{"id": id}); err != nil {
 		if errors.Is(err, mongo.ErrNoDocuments) {
-			panic("oops error occurred ")
+			return "", errors.Wrap(err, "failed to get employee")
 		}
 	}
-	return nil
+	return "employee is removed successfully", nil
 }
 
-func (c client) ListEmployee() (*[]models.Employee, error) {
-	var employee []models.Employee
-	var em models.Employee
+func (c client) ListEmployee() ([]*models.Employee, error) {
+	var employee []*models.Employee
 	collection := c.conn.Database(viper.GetString(config.DbName)).Collection(collectionName)
-	cursor, err := collection.Find(context.TODO(), bson.M{})
+	cursor, err := collection.Find(context.TODO(), bson.D{{}})
 	if err != nil {
-		log().Info("oops no records fetched")
+		return nil, errors.Wrap(err, "failed to retrieve employees")
 	}
 	defer cursor.Close(context.TODO())
 	for cursor.Next(context.TODO()) {
+		var em *models.Employee
 		if err = cursor.Decode(&em); err != nil {
-			log().Info("cannot add elements to list")
+			return nil, errors.Wrap(err, "oops error occurred while scanning.....")
```

Comment:
lets not use the word `oops`. and `.....`

Reply:
done